Documentation/Profiling/ReJIT on Attach.md

```diff
@@ -0,0 +1,68 @@
+
+```
+    HRESULT RequestReJITWithInliners(
+                [in]                       DWORD       dwRejitFlags,
+                [in]                       ULONG       cFunctions,
+                [in, size_is(cFunctions)]  ModuleID    moduleIds[],
+                [in, size_is(cFunctions)]  mdMethodDef methodIds[]);
+```
+
+Conceptually this works the same as `ICorProfilerInfo4::RequestReJIT` except it will automatically ReJIT any methods that have inlined the target method(s) in the past. The arguments are the same except for the additon of `dwRejitFlags` as the first parameter. The valid values for this argument come from this enum:
+
+``` 
+typedef enum
+{
+    // ReJITted methods will be prevented from being inlined
+    COR_PRF_REJIT_BLOCK_INLINING = 0x1,
+
+    // This flag controls whether the runtime will call GetReJITParameters
+    // on methods that are ReJITted because they inline a method that was requested
+    // for ReJIT
+    COR_PRF_REJIT_INLINING_CALLBACKS    = 0x2
+} COR_PRF_REJIT_FLAGS;
+```
+
+Any callers of this API must set `COR_PRF_REJIT_BLOCK_INLINING`. Although it is possible that in the future this restriction will be lifted, the current implementation blocks ReJITted methods from being inlined (ever).
+
+The other value `COR_PRF_REJIT_INLINING_CALLBACKS` controls whether you get a `ICorProfilerCallback4::GetReJITParameters` callback for any methods that are ReJITted as inliners of the requested method. The default is to not receive callbacks for these methods. You will always receive a `GetReJITParameters` callback for any methods that are explictly requested.
+
+
+## Inner workings/Limitations
+
+With this API you are no longer required to monitor JIT callbacks to manually block inlining from occurring. To acheive that the runtime now globally blocks a ReJITted method from being inlined (even if it was ReJITted with `ICorProfilerInfo4::RequestReJIT` and not the new API). Once a method is reverted with `ICorProfilerInfo4::RequestRevert` inlining will occur again for any future jittings.
+
+It is important to mention here how `RequestRevert` works. When you revert a ReJITted method, the original native code is activated. This means there are potential pitfalls for calling `RequestRevert`. Consider an app where method A has previously inlined both method B and method C (among potentially others), then a profiler calls `RequestReJITWithInliners` for B and C. If the profiler then wants to revert the ReJIT for method B, it might seem logical to revert method B and method A. However, when the method A code is reverted the original native code (without modification) will be activated, and that original code includes method C inlined. The result is that even though method C has been ReJITted, any future calls to method A will not call the ReJITted method C and will instead call the inlined (default IL) of method C.
```

Comment:
In this case the profiler never explicitly ReJitted A so it doesn't feel natural to me that they would try to revert it. I think this is a more precarious case:

Method A inlines method B. Profiler calls RequestReJITWithInliners on both A and B. Later profiler decides they want to get rid of the edit in A but still preserve the edit in B. The nieve implementation would be Revert(A) but that is going to run into issues because the profiler will unknowingly re-inline B and lose that edit.

I'm pretty sure once you warn them about this the immediate next question once they understand the issue is "If calling Revert(A) is bad, what should I be doing instead?"